### PR TITLE
Add intial translation of the figures for the chapters of backtracking, divide and conquer, appendix

### PR DESCRIPTION
**Pull request by krahets**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [x] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

### CODE REVIEW

**Comment by krahets** [comment]
Please review the translation based on the following aspects:

Ensure the expression is authentic and accurate.
Maintain consistency between the Chinese and English versions. (Balancing with point 1)
In addition to content, please pay attention to the format:

It is expected to use Roboto font for normal text and Roboto Mono for formulas and code. These may be prone to mistakes since they are set manually.
The style and color of shapes and texts should be consistent between the two versions. (I made some color adjustments in the EN version to enhance readability, so please only report those that feel uncomfortable to you.)

File: en/docs/chapter_backtracking/backtracking_algorithm.assets/preorder_find_constrained_paths.png
Comment (K3v123) [inline en/docs/chapter_backtracking/backtracking_algorithm.assets/preorder_find_constrained_paths.png:1]: "pruning" is not red. and the value 3 is red (which i like).

File: en/docs/chapter_backtracking/n_queens_problem.assets/n_queens_constraints.png
Comment (K3v123) [inline en/docs/chapter_backtracking/n_queens_problem.assets/n_queens_constraints.png:1]: EN RHS the text are not bold

File: en/docs/chapter_backtracking/n_queens_problem.assets/n_queens_placing.png
Comment (K3v123) [inline en/docs/chapter_backtracking/n_queens_problem.assets/n_queens_placing.png:1]: EN text are not bold

File: en/docs/chapter_backtracking/permutations_problem.assets/permutations_i_pruning.png
Comment (K3v123) [inline en/docs/chapter_backtracking/permutations_problem.assets/permutations_i_pruning.png:1]: just EN not bold for round

File: en/docs/chapter_backtracking/permutations_problem.assets/permutations_i.png
Comment (K3v123) [inline en/docs/chapter_backtracking/permutations_problem.assets/permutations_i.png:1]: below image the rounds are in bold but this one both CN and en are not bold

File: en/docs/chapter_backtracking/permutations_problem.assets/permutations_ii.png
Comment (K3v123) [inline en/docs/chapter_backtracking/permutations_problem.assets/permutations_ii.png:1]: same not consistent.
also the "should eliminate duplicate...." is not bold

File: en/docs/chapter_backtracking/permutations_problem.assets/permutations_ii_pruning.png
Comment (K3v123) [inline en/docs/chapter_backtracking/permutations_problem.assets/permutations_ii_pruning.png:1]: pruning color suppose to be red
and text should be bold for the red texts

File: en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i_pruning.png
Comment (K3v123) [inline en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i_pruning.png:1]: "pruning" are bold in CN but in EN its not bold and the text are grey but en are not grey

File: en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_ii.png
Comment (K3v123) [inline en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_ii.png:1]: text not grey.
"target" suppose to be green

File: en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_divide_and_conquer.png
Comment (K3v123) [inline en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_divide_and_conquer.png:1]: im not sure if its just me, the color grey and black is a bit hard to be determined compared to the text style from cn

File: en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_f3_step1.png
Comment (K3v123) [inline en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_f3_step1.png:1]: color not match, this is all red. CN is grey and green. and the text dont fit the grey box in EN!!!!

**Review by K3v123 — COMMENTED** [review]
overall looks good to me.
there is one part that the bold and black text difference is a bit hard to be determined because of the style of text. and there is mismatch in colors in some other ones.

File: en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i.png
Comment (pengchzn) [inline en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i.png:1]: “Element sum cannot..."is some kind direct translation, "The sum of elements cannot exceed target." may be better

File: en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i_pruning.png
Comment (pengchzn) [reply en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i_pruning.png:1]: **Subset** in the left column should be on the same line

File: en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_ii.png
Comment (pengchzn) [reply en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_ii.png:1]: “Element sum cannot..."is some kind direct translation, "The sum of elements cannot exceed target." may be better

Comment (pengchzn) [reply en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_ii.png:1]: "Pruning 3: Do not allow...elements" -> "Duplicate selection of array elements is not allowed"

File: en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_divide_and_conquer.png
Comment (pengchzn) [reply en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_divide_and_conquer.png:1]: ”Subproblem f(1): ... to C“ -> ”Move remaining 1 disk from A to C“

File: en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_f3_step1.png
Comment (pengchzn) [reply en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_f3_step1.png:1]: Treat the top two disks as a whole.

File: en/docs/chapter_backtracking/backtracking_algorithm.assets/backtrack_remove_return_or_not.png
Comment (yuelinxin) [inline en/docs/chapter_backtracking/backtracking_algorithm.assets/backtrack_remove_return_or_not.png:1]: might be better to replace "with" with a comma.

File: en/docs/chapter_backtracking/permutations_problem.assets/permutations_i_pruning.png
Comment (yuelinxin) [reply en/docs/chapter_backtracking/permutations_problem.assets/permutations_i_pruning.png:1]: > "1st round choice **1**" -> "1st round choose **1**", it should be verb, not noun. Same with 2nd and 2rd round

yes

File: en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i_pruning.png
Comment (yuelinxin) [reply en/docs/chapter_backtracking/subset_sum_problem.assets/subset_sum_i_pruning.png:1]: > **Subset** in the left column should be on the same line

true

File: en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_f3_step1.png
Comment (yuelinxin) [reply en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_f3_step1.png:1]: yes the styling is off

**Review by yuelinxin — COMMENTED** [review]
There are some wording and styling issues, but amazing work! Please consider my suggestions.

File: en/docs/chapter_backtracking/backtracking_algorithm.assets/backtrack_remove_return_or_not.png
Comment (krahets) [reply en/docs/chapter_backtracking/backtracking_algorithm.assets/backtrack_remove_return_or_not.png:1]: Record solution, then continue searching without returning

File: en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_divide_and_conquer.png
Comment (krahets) [reply en/docs/chapter_divide_and_conquer/hanota_problem.assets/hanota_divide_and_conquer.png:1]: > ”Subproblem f(1): ... to C“ -> ”Move remaining 1 disk from A to C“
No space for this

**Pull request merged**